libs/agno/agno/os/interfaces/a2a/router.py

```diff
@@ -239,6 +246,124 @@ async def a2a_stream_agent(request: Request, id: str):
         except Exception as e:
             raise HTTPException(status_code=500, detail=f"Failed to start run: {str(e)}")
 
+    @router.get(
+        "/agents/{id}/v1/tasks",
+        operation_id="get_agent_task",
+        name="get_agent_task",
+        description="Retrieve an A2A Task by its ID.",
+        response_model_exclude_none=True,
+        responses={
+            200: {
+                "description": "Task retrieved successfully",
+                "content": {
+                    "application/json": {
+                        "example": {
+                            "jsonrpc": "2.0",
+                            "id": "request-123",
+                            "result": {
+                                "id": "task-456",
+                                "context_id": "context-789",
+                                "status": {"state": "completed"},
+                                "history": [
+                                    {
+                                        "message_id": "msg-1",
+                                        "role": "agent",
+                                        "parts": [{"kind": "text", "text": "Response from agent"}],
+                                    }
+                                ],
+                            },
+                        }
+                    }
+                },
+            },
+            404: {"description": "Task not found"},
+        },
+    )
+    async def get_task_agent(
+        request: Request,
+        id: str,
+        session_type: Optional[SessionType] = None,
+        user_id: Optional[str] = None,
+    ):
+        """Retrieve a task by ID. This endpoint returns an A2A Task object."""
+        request_body = await request.json()
+        request_id = request_body.get("id", "unknown")
+        task_id = request_body.get("params", {}).get("task_id")
+        session_id = request_body.get("params", {}).get("session_id")
```

Comment:
This is the endpoint to get the list of tasks for the given agent (`/agents/{id}/v1/tasks`), so:

- We don't need `task_id` in the request body
- Let's rename the function to `get_agent_tasks`